src/test_driver/linux-cirque/helper/CHIPTestBase.py

```diff
@@ -0,0 +1,287 @@
+
+        self.logger.info("wifi ap ssid: {}, psk: {}".format(ssid, psk))
+
+        for device in self.non_ap_devices:
+            self.logger.info(
+                "device id: {} connecting to desired ssid: {}".format(
+                    device.device_id, ssid))
+            write_psk_to_wpa_supplicant_config(
+                self.logger, self.stub, self.home_id, device.device_id, ssid, psk)
+
+            kill_existing_wpa_supplicant(
+                self.logger, self.stub, self.home_id, device.device_id)
+
+            start_wpa_supplicant(
+                self.logger, self.stub, self.home_id, device.device_id)
+        time.sleep(5)
+
+    def get_device_thread_ip(self, device_id):
+        ret = self.stub.ExecuteDeviceCommand(
+                service_pb2.ExecuteDeviceCommandRequest(
+                    home_id=self.home_id,
+                    device_id=device_id,
+                    command="ot-ctl ipaddr",
+                    streaming=False))
+        ipaddr_list = ret.output.splitlines()
+        for ipstr in ipaddr_list:
+            try:
+                self.logger.info("device id: {} thread ip: {}".format(device_id, ipstr))
+                ipaddr = ipaddress.ip_address(ipstr)
+                if ipaddr.is_link_local:
+                    self.logger.info("ignore ip: {} : link local".format(ipstr))
+                    continue
+                if not ipaddr.is_private:
+                    self.logger.info("ignore ip: {} : global ip addr".format(ipstr))
+                    continue
+                if re.match("fd[0-9a-f]{2}:[0-9a-f]{4}:[0-9a-f]{4}:[0-9a-f]{4}:0000:00ff:fe00:[0-9a-f]{4}", ipaddr.exploded) != None:
+                    self.logger.info("ignore ip: {} : not for app".format(ipstr))
+                    continue
+                self.logger.info("return ip: {}".format(ipstr))
+                return str(ipaddr)
+            except:
+                pass
+        return None
+    
+    def get_device_log(self, device_id, tail=0):
+        if tail == 0:
+            # Cirque contains a bug that it cannot pass all to docker since tail is a uint32
+            # Set it to INT32_MAX for maximum lines of logs
+            tail = 2147483647
+        return self.stub.GetCirqueDeviceLog(service_pb2.GetCirqueDeviceLogRequest(
+            home_id=self.home_id, device_id=device_id, tail=tail))
+        
+    def save_device_logs(self):
+        timestamp = int(time.time())
+        if not os.path.exists("logs"):
+            os.makedirs("logs")
+
+        for device in self.non_ap_devices:
+            ret_log = self.get_device_log(device.device_id)
+            f_name = '{}-{}-{}.log'.format(device.device_specification.device_type, timestamp, device.device_id[-8:])
+            self.logger.debug("device log name: \n{}".format(f_name))
+            with open(os.path.join('logs', f_name), 'w') as fp:
+                fp.write(ret_log.log)
+
+
+def create_device_from_config():
```

Comment:
could we use DEVICE_CONFIG.values()  insead?

I can only see the config used as a list - should it be a list instead of a dict?